modules/goldilocks/main.tf

```diff
@@ -0,0 +1,102 @@
+locals {
+  auth = jsonencode({ "userPoolARN" = "${var.userpoolarn}", "userPoolClientID" = "${var.userpoolclientid}", "userPoolDomain" = "${var.userpooldomain}" })
+}
+
+data "aws_route53_zone" "domain-zone" {
+  name = var.zone_name
+}
+
+resource "helm_release" "vpa" {
+  count      = var.create_vpa_server ? 1 : 0
+  name       = "goldilocks"
+  version    = "0.5.0"
+  repository = "https://charts.fairwinds.com/stable"
+  chart      = "vpa"
+}
+
+resource "helm_release" "metric_server" {
+  count      = var.create_metric_server ? 1 : 0
+  name       = "metrics-server"
+  version    = "5.8.5"
+  repository = "https://kubernetes-sigs.github.io/metrics-server/"
+  chart      = "metrics-server"
+  namespace  = "kube-system"
+}
+
+resource "kubernetes_manifest" "create_namespace" {
+  manifest = {
+    "apiVersion" = "v1"
+    "kind"       = "Namespace"
+    "metadata" = {
+      "name" = "goldilocks"
+    }
+  }
+}
+
+resource "null_resource" "vpa_configure" {
```

Comment:
please add also destroy provisioner and triggers depending on namespaces,
not sure if destroy provisioner is possible(here is the doc https://discuss.hashicorp.com/t/how-to-rewrite-null-resource-with-local-exec-provisioner-when-destroy-to-prepare-for-deprecation-after-0-12-8/4580/7) but triggers is possible to set to run in case if we add additional namespaces after apply